plugins/module_utils/utils.py

```diff
@@ -26,3 +28,103 @@ def generate_api_endpoint(path, **kwargs):
+            "bfdMultiHopPol": {
+                "adminState": "enabled",
+                "detectionMultiplier": 3,
+                "minRxInterval": 250,
+                "minTxInterval": 250
+            },
+        }
+        update_path = "/tenantPolicyTemplate/template/l3OutIntfPolGroups/0"
+
+        Sample existing_data_keys input format:
+            To update the "bfdMultiHopPol -> adminState" value:
+                existing_data_keys = "bfdMultiHopPol", "adminState"
+
+        To update the top level key of the existing object:
+            append_update_ops_data(ops, existing_data, update_path, "L3OutInterfacePolicy_NameUpdated", "name", op="replace")
+
+        To create the nested object from the scratch when the object is not available:
+            append_update_ops_data(ops, existing_data, update_path, dict(), "ospfIntfPol", op="replace")
+            append_update_ops_data(ops, existing_data, update_path, dict(), "ospfIntfPol", "ifControl", op="replace")
+            append_update_ops_data(ops, existing_data, update_path, True, "ospfIntfPol", "ifControl", "ignoreMtu", op="replace")
+
+        After inserting the new object:
+        existing_data = {
+            "name": "L3OutInterfacePolicy",
+            "ospfIntfPol": {
+                "networkType": "broadcast",
+                "prio": 1,
+                "cost": 0,
+                "ifControl": {
+                    "advertiseSubnet": false,
+                    "bfd": false,
+                    "ignoreMtu": false,
+                    "passiveParticipation": false
+                },
+                "helloInterval": 10,
+                "deadInterval": 40,
+                "retransmitInterval": 5,
+                "transmitDelay": 1
+            },
+        }
+
+        To update the nested key of the existing object:
+            append_update_ops_data(ops, existing_data, update_path, True, "ospfIntfPol", "ifControl", "ignoreMtu", op="replace")
+
+        To delete the nested object from the existing object:
+            append_update_ops_data(ops, existing_data, update_path, None, "ospfIntfPol", "ifControl", op="remove")
+
+        To delete the top level key from the existing object:
+            append_update_ops_data(ops, existing_data, update_path, None, "ospfIntfPol", op="remove")
+    """
+
+    def recursive_update(data, path, keys, new_value):
+        if len(keys) == 0:
+            return
+        key = keys[0]
+
+        if len(keys) == 1:
+            # Update the existing configuration
+            if new_value is not None and data.get(key) != new_value and op == "replace":
+                data[key] = new_value
+                ops.append(
+                    dict(
+                        op=op,
+                        path="{}/{}".format(path, key),
+                        value=copy.deepcopy(new_value),
+                    )
+                )
+            # Clear the existing configuration
+            elif op == "remove" and key in data:
```

Comment:
We can throw error when the input is empty and `op` is not `replace` or `remove`.
Can we return the error message or fail the module execution?

Reply:
Maybe it should fail the module. It will most likely be an issue when modules are first developed and testing would quickly pick up the failure.
Does throwing an exception get caught as a module failure in Ansible?

Reply:
@samiib It does get caught and the module will therefore fail. I think we should throw an error as well just in case:
```Python
valid_ops = ["replace", "remove"]
if op not in valid_ops:
  raise ValueError("Invalid op value. Expected one of: {0}. Got: {1}".format(valid_ops, op))
```